pkg/hashivault/auth/auth.go

```diff
@@ -45,7 +45,7 @@ func Authenticate(ctx context.Context, addr string, method Method, opts ...Optio
 
 	switch method {
 	case MethodOICD:
-		return authOICD(spanCtx, addr)
+		return authOICD(spanCtx, addr, newCache())
```

Comment:
The cache should be created from the config/options.

Now, one can't choose not to have it stored locally, and if there are any issues will become a blocker. Also, if one wants to "reset" the cache for any reason (eg testing), one would need to find the stored file and delete it, by knowing the path and filename.

Users of the lib should have the option to supply a "noop" or mock cache if needed.